### PR TITLE
Improve readiness tags

### LINKED ISSUE #317: Enhance GPopper
Enhance the Vue-Popper wrapper to have dedicated slots for different kind of messages. We should also remove the `message`property in favor for slots. Moreover, we should leverage Xterm.js to display log messages with proper formatting (including colors and other Ascii symbols).

### LINKED ISSUE #518: Support new Conditions
Support new Conditions:
- `GardenletReady`(`GR`)
- `Bootstrapped` (`B`)
- `AuditlogServiceAvailability` (`ASA`)

The Conditions show up for shooted seeds which are only visible to operators already

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:
Support new Conditions

**Which issue(s) PR fixes**:
Fixes #518 Fixes #317

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Added possibility to define names and descriptions for custom conditions in dashboard config and allow to overwrite existing ones
```
```improvement user
ANSI sequences in error messages are now rendered properly
```

### CODE REVIEW

File: frontend/src/components/ShootStatus.vue [hunk 1]
```diff
@@ -61,6 +61,9 @@ import map from 'lodash/map'
 import join from 'lodash/join'
 import { isUserError } from '@/utils'
 
+const ansiHTML = require('ansi-html')
+const escape = require('escape-html');
```
Comment (holgerkoser) [inline frontend/src/components/ShootStatus.vue]: import ansiHTML from 'ansi-html'
import escape from 'lodash/escape'

Comment (grolu) [reply frontend/src/components/ShootStatus.vue]: done

File: frontend/src/components/GPopper.vue [hunk 2]
```diff
@@ -21,6 +21,8 @@ limitations under the License.
     trigger="click"
     :options="popperOptions"
     boundaries-selector=".v-content__wrap"
+    @show="onPopperShow"
```
Comment (holgerkoser) [inline frontend/src/components/GPopper.vue]: tooltip above gpopper (z-index)

File: frontend/package.json [hunk 3]
```diff
@@ -22,12 +22,14 @@
...
   "dependencies": {
     "@mdi/font": "^4.4.95",
+    "ansi-html": "^0.0.7",
     "axios": "^0.19.0",
     "clipboard": "^2.0.1",
     "codemirror": "^5.40.2",
     "component-emitter": "^1.2.1",
     "dompurify": "^2.0.7",
     "downloadjs": "^1.4.7",
+    "escape-html": "^1.0.3",
```
Comment (holgerkoser) [inline frontend/package.json]: no needed. We use lodash.

File: frontend/src/components/GPopper.vue (hunk 2 above)
Comment (holgerkoser) [reply frontend/src/components/GPopper.vue]: forward events remove value prop

File: frontend/src/components/MemberServiceAccountsRow.vue [hunk 4]
```diff
@@ -32,14 +32,14 @@ limitations under the License.
             {{displayName}}
           </v-list-tile-title>
           <v-layout row
-            slot="content-before"
+            slot="content"
```
Comment (holgerkoser) [inline frontend/src/components/MemberServiceAccountsRow.vue]: make content slot the default slot

File: frontend/src/components/ShootStatus.vue [hunk 5]
```diff
@@ -186,7 +189,7 @@ export default {
       if (message === this.lastErrorDescription) {
         return undefined
       }
-      return message
+      return ansiHTML(escape(message))
```
Comment (holgerkoser) [inline frontend/src/components/ShootStatus.vue]: make function escapedAnsihtml

File: frontend/src/components/StatusTag.vue [hunk 6]
```diff
@@ -17,51 +17,57 @@ limitations under the License.
 <template>
   <span v-if="visible">
     <template v-if="tag.message">
-      <g-popper @rendered="popperRendered=true" :title="chipTitle" :message="tag.message" :toolbarColor="color" :time="{ caption: 'Last updated:', dateTime: tag.lastUpdateTime }" :popperKey="popperKeyWithType" :placement="popperPlacement">
-        <v-tooltip slot="popperRef" top>
+      <g-popper v-model="popperVisible" @rendered="popperRendered=true" :title="chipTitle" :toolbarColor="color" :time="{ caption: 'Last updated:', dateTime: tag.lastUpdateTime }" :popperKey="popperKeyWithType" :placement="popperPlacement">
```
Comment (holgerkoser) [inline frontend/src/components/StatusTag.vue]: change to @input

File: frontend/src/components/ShootStatus.vue [hunk 7]
```diff
@@ -40,15 +40,15 @@ limitations under the License.
         <div>{{ tooltipText }}</div>
       </v-tooltip>
     </div>
-    <template slot="content-after">
-      <pre v-if="!!popperMessage" class="alert-message">{{ popperMessage }}</pre>
+    <template slot="content">
+      <pre v-if="!!popperMessage" class="alert-message" v-html="popperMessage"></pre>
```
Comment (holgerkoser) [inline frontend/src/components/ShootStatus.vue]: may be an extra component for ansiHTML display?

File: frontend/src/components/StatusTag.vue [hunk 8]
```diff
@@ -17,51 +17,57 @@ limitations under the License.
...
       </g-popper>
     </template>
-    <template v-else>
+    <template v-else max-width="400px">
       <v-tooltip top>
         <v-chip class="status-tag" slot="activator" outline :text-color="chipTextColor" small :color="color">
           {{chipText}}
         </v-chip>
-        <span>{{chipTooltip}}</span>
+        <span v-html="chipTooltip"></span>
```
Comment (holgerkoser) [inline frontend/src/components/StatusTag.vue]: inline html into template

File: frontend/src/components/StatusTag.vue [hunk 9]
```diff
@@ -187,7 +203,34 @@ export default {
...
       return `${name} [${errorState}${since}]`
+    },
+    conditionFromType (type) {
+      let condition = knownConditions[type]
+      if (condition) {
+        return condition
+      }
+      let displayNameComponents = []
+      let shortNameComponents = []
+      const conditionPattern = /[A-Z]*([A-Z])[a-z]*/g
```
Comment (holgerkoser) [inline frontend/src/components/StatusTag.vue]: use snakeCase and whitelist for suffix

File: frontend/src/components/StatusTag.vue [hunk 10]
```diff
@@ -187,7 +203,34 @@ export default {
...
+      }
+
+      const displayName = join(displayNameComponents, ' ')
+      const shortName = join(shortNameComponents, '')
+      knownConditions[type] = { displayName, shortName } // cache
+      return knownConditions[type]
     }
+  },
+  beforeMount () {
+    merge(knownConditions, this.cfg.knownConditions)
```
Comment (holgerkoser) [inline frontend/src/components/StatusTag.vue]: move knownConditions to store and rename known to wellKnown?

File: frontend/tests/unit/StatusTag.spec.js [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+      },
+      modules: {},
+      getters: {
+        isAdmin: () => {
+          return false
+        }
+      }
+    })
+  })
+  it('should generate condition object for unknown condition types', function () {
```
Comment (holgerkoser) [inline frontend/tests/unit/StatusTag.spec.js]: split into multiple tests and may be reuse function

File: frontend/src/components/GPopper.vue (hunk 2 above)
Comment (grolu) [reply frontend/src/components/GPopper.vue]: done

File: frontend/src/components/MemberServiceAccountsRow.vue (hunk 4 above)
Comment (grolu) [reply frontend/src/components/MemberServiceAccountsRow.vue]: done

File: frontend/src/components/ShootStatus.vue (hunk 7 above)
Comment (grolu) [reply frontend/src/components/ShootStatus.vue]: done

File: frontend/src/components/StatusTag.vue (hunk 8 above)
Comment (grolu) [reply frontend/src/components/StatusTag.vue]: done

File: frontend/src/components/StatusTag.vue (hunk 9 above)
Comment (grolu) [reply frontend/src/components/StatusTag.vue]: done

File: frontend/src/components/StatusTag.vue (hunk 10 above)
Comment (grolu) [reply frontend/src/components/StatusTag.vue]: done

File: frontend/src/components/AnsiText.vue [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+</template>
+
+<script>
+import ansiHTML from 'ansi-html'
+import escape from 'lodash/escape'
+
+export default {
+  props: {
+    text: {
+      type: String
```
Comment (holgerkoser) [inline frontend/src/components/AnsiText.vue]: default empty string

File: frontend/src/components/StatusTag.vue [hunk 13]
```diff
@@ -16,26 +16,41 @@ limitations under the License.
...
         </v-chip>
-        <span v-html="chipTooltip"></span>
+        <span class="font-weight-bold">{{chipTooltip.title}}</span>
+        <div v-if="chipTooltip.description">
+          {{chipTooltip.description}}
+        </div>
       </v-tooltip>
-    </template>
-    <time-string v-if="popperRendered" v-show="false" :dateTime="tag.lastTransitionTime" :currentString.sync="lastTransitionString" :pointInTime="-1" :withoutPrefixOrSuffix="true"></time-string>
+      <ansi-text :text="tag.message" :class="{ 'error--text': isError }"></ansi-text>
```
Comment (holgerkoser) [inline frontend/src/components/StatusTag.vue]: not red text if error

File: frontend/src/components/StatusTag.vue [hunk 14]
```diff
@@ -156,7 +140,7 @@ export default {
     tag () {
       const { lastTransitionTime, lastUpdateTime, message, status, type } = this.condition
       const id = type
-      const { displayName: name, shortName, description } = this.conditionFromType(type)
+      const { displayName: name, shortName, description } = this.conditionMetaDataFromType(type)
```
Comment (holgerkoser) [inline frontend/src/components/StatusTag.vue]: Metadata

File: frontend/src/components/StatusTag.vue [hunk 15]
```diff
@@ -175,7 +159,8 @@ export default {
     visible () {
       if (!this.isAdmin) {
         const { type } = this.condition
-        return !get(knownConditions, [type, 'showAdminOnly'], false)
+
+        return !get(this.conditionMetaDataFromType(type), 'showAdminOnly', false)
```
Comment (holgerkoser) [inline frontend/src/components/StatusTag.vue]: extract variable

File: frontend/src/components/StatusTag.vue [hunk 16]
```diff
@@ -208,36 +196,35 @@ export default {
...
+      if (includes(dropSuffixes, last(conditionComponents))) {
+        conditionComponents = dropRight(conditionComponents)
       }
 
-      const displayName = join(displayNameComponents, ' ')
-      const shortName = join(shortNameComponents, '')
-      knownConditions[type] = { displayName, shortName } // cache
-      return knownConditions[type]
+      const displayName = join(conditionComponents, ' ')
+      const shortName = join(map(conditionComponents, conditionComponent => conditionComponent.charAt(0)), '')
```
Comment (holgerkoser) [inline frontend/src/components/StatusTag.vue]: first

File: frontend/tests/unit/StatusTag.spec.js [hunk 17]
```diff
@@ -30,157 +31,95 @@ Object.defineProperties(localVue.prototype, {
...
+    expect(statusTag.description).to.equal(undefined)
+  })
+
+  it('should cache generated condition object for unknown condition type', function () {
+    expect(store.state.conditionCache.UnknownCondition).to.equal(undefined)
+
+    const statusTag = createStatusTagComponent('UnknownCondition')
+    expect(statusTag.shortName).to.equal('UC')
+
+    expect(store.state.conditionCache.UnknownCondition).to.not.equal(undefined)
```
Comment (holgerkoser) [inline frontend/tests/unit/StatusTag.spec.js]: test eql object

**Pull request merged**